### ISSUE TITLE
Finalize output requirements

### ISSUE THREAD

**Issue by RachaelDMueller** [MIDOSS/MIDOSS-MOHID-config#7]
Listed below is our current set of outputs.  

**1. Single point (scalar) values to calculate and/or put in model output “stats” file**
- earliest beaching time 
- mode of beaching time histogram based on 1/2 day interval (time value) _(actually mode of first beaching time)_
- last value of mass evaporated 
- last value of mass dispersed _(change to maximum value of mass dispersed)_
- last value of mass of surface oil (original (volume * density at first time step in .sro) - total mass beached - total mass weathered (dispersed, biodegrade, evaporated), at last time step _(why not just use the mass of surface oil directly?)_
- Oil-affected area (birds): Cumulative area with oil thickness greater than zero 
- Oil-affected area (spill cleanup): Cumulative area with oil thickness greater than 20 mm 
- Spill area at last time step
- Oil spread: max distance between oil particles over course of model run
- Oil advection: max distance from spill site that oil reaches, over course of model run
- mode of oil spill distance (from surface oil, not beached)
- (single value) total mass beached (density(in .sro file) * 500 * 440 * thickness2D(in .nc) * 1e-6)  _(why not just used the volume beached as output in the sro file?)_

**2. Mappable values to calculate and add to output netcdf**
- 2D (spatial representation) of mass and concentration of beached oil at last time step
- 2D map of beaching time scales
- 2D map of floating-oil-affected area (stakeholder: birds) as represented as a binary flag [0,1] for oil [not present, present] at each time step that is then time integrated and divided by total time steps to represent the fraction of time over model run in which a particular grid cell is affected by oil
- 2D map of floating-oil-affected area (stakeholder: responders) as represented as a binary flag [0,1] for oil [not present and/or < 20 mm, present at >=20 mm] at each time step that is then time integrated and divided by total time steps to represent the fraction of time over model run in which a particular grid cell is affected by oil

**3. Values already present in output netcdf**
- Thickness_2D (time, grid_y, grid_x)
- OilConcentration_3D (time, grid_z, grid_y, grid_x)
- Beaching_Time (grid_y, grid_x)

**4. Values already present in output netcdf that I don’t yet trust and wouldn’t use w/o vetting**
- OilConcentration_2D  (time, grid_y, grid_x) - If I recall correctly, the values in this are garbage
- Dissolution_3D  (time, grid_z, grid_y, grid_x)
- Beaching_Volume (grid_y, grid_x) - These values are dates not volume. 
- Oil_Arrival_Time (grid_y, grid_x) - Not clear to me yet how these values differ from “Beaching_Time”.  I looked into this at one point and need to review/revisit.

**Comment by SusanEAllen** [comment]
"Oil-affected area: Cumulative area with oil thickness above a certain threshold (start with zero)"
I think we actually want two of these.  One for sea-birds and one for clean up. 
zero is good for seabirds
I think 20-25 mm for skimmers.  Based on needed 50 mm (https://www.bsee.gov/sites/bsee.gov/files/osrr-oil-spill-response-research/osrr-1072aa.pdf) and the idea that booms could concentrate the thickness by a factor of 2 or 2.5

**Comment by RachaelDMueller** [comment]
Good idea, Susan. I’ve update the above list to reflect this differentiation in spill area calculations.  I’ve also specified total mass of beached oil to be calculated as:

 mass = density(at last time step in .sro file) * 500 * 440 * thickness2D(in .nc) * 1e-6 

This way, we avoid the need to incorporate changes in e3t and the calculation is more straightforward.

**Comment by RachaelDMueller** [comment]
I’m also updating the google spreadsheet that we have been using to sketch out comparisons and track progress on this front.  See “Variable Comparisons” in [this spreadsheet](https://docs.google.com/spreadsheets/d/1PN_YcgsmmDJKlPP1SQlbD7_DI1BoNRHvbm87bPZYD2M/edit#gid=2053290476) for more information.

**Comment by RachaelDMueller** [comment]
With spill spread and advection, need to calculate based on every half hour b/c last time step may not accurately reflect max distances

**Comment by RachaelDMueller** [comment]
1. Single point (scalar) values to calculate and/or put in model output “stats” file
earliest beaching time
mode of beaching time histogram based on 1/2 day interval (time value)
last value of mass evaporated
peak value of mass dispersed
last value of mass of surface oil (original (volume * density at first time step in .sro) - total mass beached - total mass weathered (dispersed, biodegrade, evaporated), at last time step
Oil-affected area (birds): Cumulative area with oil thickness greater than zero
Oil-affected area (spill cleanup): Cumulative area with oil thickness greater than 20 mm
Spill area at last time step
Oil spread: max distance between oil particles over course of model run
Oil advection: max distance from spill site that oil reaches, over course of model run
mode of oil spill distance (from surface oil, not beached)
(single value) total mass beached (density(in .sro file) * 500 * 440 * thickness2D(in .nc) * 1e-6)

2. Mappable values to calculate and add to output netcdf
2D (spatial representation) of volume of beached oil at last time step
2D map of beaching time scales
2D map of floating-oil-affected area (stakeholder: birds) as represented as a binary flag [0,1] for oil [not present, present] at each time step that is then time integrated and divided by total time steps to represent the fraction of time over model run in which a particular grid cell is affected by oil
2D map of floating-oil-affected area (stakeholder: responders) as represented as a binary flag [0,1] for oil [not present and/or < 20 mm, present at >=20 mm] at each time step that is then time integrated and divided by total time steps to represent the fraction of time over model run in which a particular grid cell is affected by oil
3. Values already present in output netcdf

Thickness_2D (time, grid_y, grid_x)  - NEED TO EVALUATE
OilConcentration_3D (time, grid_z, grid_y, grid_x)
Beaching_Time (grid_y, grid_x)

4. Values already present in output netcdf that I don’t yet trust and wouldn’t use w/o vetting

OilConcentration_2D (time, grid_y, grid_x) - If I recall correctly, the values in this are garbage
Dissolution_3D (time, grid_z, grid_y, grid_x)
Oil_Arrival_Time (grid_y, grid_x) - Not clear to me yet how these values differ from “Beaching_Time”. I looked into this at one point and need to review/revisit.

**Comment by RachaelDMueller** [comment]
@SusanEAllen and @campower,  In[20] in [this notebook](https://nbviewer.jupyter.org/github/MIDOSS/analysis-rachael/blob/main/notebooks/OS2020/SB_akns_3panel_probability.ipynb) from our OS2020 poster is the kind of graphic that I had in mind for "2D map of beaching time scales.”  If we go this way, we will need to consider what to do with cases where beaching occurs at depth.  My thinking here is to mask those locations out, though we may want to track the volume of "beaching at depth" vs. shoreline, in particular for the Bunker-C case, for issues of transparency and mass balance comparisons.
